File: frontend/packages/library/src/severity.ts
// SOCP 通用工具库：级别颜色映射（供各 app 复用）
export const SEVERITY_COLORS: Record<string, string> = {
  // Framework-independent fallback palette. The workbench resolves the same
  // semantic tones from CSS variables at runtime for light/dark themes.
  CRITICAL: '#dc2626',
  HIGH: '#dc2626',
  MEDIUM: '#b45309',
  LOW: '#667085',
  INFO: '#667085',
}

export function severityColor(level: string): string {
  return SEVERITY_COLORS[(level || 'INFO').toUpperCase()] ?? SEVERITY_COLORS.INFO
}

File: platform/socp-rule/src/main/java/com/socp/rule/model/Severity.java
package com.socp.rule.model;

/**
 * 告警/事件严重级别。数值越大越严重，便于比较。由 com.siem 迁移。
 */
public enum Severity {
    INFO(0),
    LOW(1),
    MEDIUM(2),
    HIGH(3),
    CRITICAL(4);

    private final int level;

    Severity(int level) {
        this.level = level;
    }

    public int level() {
        return level;
    }

    /** 是否达到（>=）给定的最低严重级别 */
    public boolean atLeast(Severity other) {
        return this.level >= other.level;
    }

    public static Severity fromLevel(int level) {
        for (Severity s : values()) {
            if (s.level == level) return s;
        }
        return level >= CRITICAL.level ? CRITICAL : INFO;
    }
}
